core/server/storage/local-file-store.js

```diff
@@ -51,14 +53,52 @@ LocalFileStore.prototype.exists = function (filename) {
 };
 
 // middleware for serving the files
-LocalFileStore.prototype.serve = function () {
-    // For some reason send divides the max age number by 1000
-    // Fallthrough: false ensures that if an image isn't found, it automatically 404s
-    return serveStatic(config.paths.imagesPath, {maxAge: utils.ONE_YEAR_MS, fallthrough: false});
+LocalFileStore.prototype.serve = function (options) {
+    var self = this;
+    options = options || {};
+
+    // CASE: serve themes
+    // serveStatic can't be used to serve themes, because
+    // download files depending on the route (see `send` npm module)
+    if (options.isTheme) {
```

Comment:
When uploading a theme, we unzip via gscan and delete the zip file from multer here https://github.com/TryGhost/Ghost/pull/7209/files#diff-8d96493a641ecbe40ffd7f378b4a175fR108.
So only on download we generate the zip if it doesn't exists. See reasoning in first comment.